scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/EquationSolvingAlgorithms.java

```diff
@@ -0,0 +1,417 @@
+    public NoSolutionException(String errorMessage) {
+      super(errorMessage);
+    }
+  }
+
+  final static class BrentDoubleAlgorithm implements BracketingAlgorithm<Double> {
+    public RootFindingResult<Double> findRoot(
+        Function<Double> f,
+        Double bracket1,
+        Double bracket2,
+        Double y,
+        Double confidenceYLow,
+        Double confidenceYHigh,
+        Double dxLow,
+        Double dxHigh,
+        int maxNbIterations)
+    throws ExceededMaxIterationException, WrongBracketingException, NoSolutionException
+    {
+      if (bracket1 < dxLow || bracket2 > dxHigh) {
+        throw new WrongBracketingException("brackets are out of prescribed root domain");
+      }
+      if (Math.abs(confidenceYLow) - Math.abs(confidenceYHigh) < 1E-6) {
+        throw new NoSolutionException("To use Brent's method, you must provide equal lower and upper confidence bounds.");
+      }
+      BrentSolver solver = new BrentSolver(confidenceYHigh);
+      //the brent implementation solves f(x) = 0 so we provide lf(x) = f(x) - y
+      var lf = new UnivariateFunction() {
+        @Override
+        public double value(final double v) {
+          return f.valueAt(v) - y;
+        }
+      };
+      try {
+        final var root = solver.solve(maxNbIterations, lf, bracket1, bracket2);
+        return new RootFindingResult<>(root, f.valueAt(root),null);
+      } catch (TooManyEvaluationsException e) {
+        throw new ExceededMaxIterationException();
+      } catch (NoBracketingException e) {
+        throw new WrongBracketingException(e.getMessage());
+      }
+    }
+  }
+  final static class BisectionDoubleAlgorithm implements BracketingAlgorithm<Double> {
+
+    public RootFindingResult<Double> findRoot(
+        Function<Double> f,
+        Double bracket1,
+        Double bracket2,
+        Double y,
+        Double confidenceYLow,
+        Double confidenceYHigh,
+        Double dxLow,
+        Double dxHigh,
+        int maxNbIterations)
+    throws WrongBracketingException, ExceededMaxIterationException
+    {
+      if (bracket1 < dxLow || bracket1 > dxHigh || bracket2 < dxLow || bracket2 > dxHigh) {
+        throw new WrongBracketingException("Brackets are out of prescribed root domain");
+      }
+      final var ff = new Function<Double>(){
+      @Override
+      public Double valueAt(final Double x) {
+        return f.valueAt(x) - y;
+      }
+
+      @Override
+      public boolean isApproximation() {
+        return f.isApproximation();
+      }
+    };
+
+      double fx_m = ff.valueAt(bracket1);
+      double fx_p = ff.valueAt(bracket2);
+      if (fx_m * fx_p > 0) {
+        throw new WrongBracketingException("f(x0)=" + fx_m + " and f(x1)=" + fx_p + " are of the same sign");
+      }
+      for (int nbIt = 0; nbIt < maxNbIterations; nbIt++) {
+        final double xg = (bracket1 + bracket2) / 2.;
+        final var fxg = ff.valueAt(xg);
+        if (fxg >= confidenceYLow && fxg <= confidenceYHigh) {
```

Comment:
Should this be minus `confidenceYLow`?:
```suggestion
        if (fxg >= -confidenceYLow && fxg <= confidenceYHigh) {
```

Reply:
Good catch. I also harmonized how theses bounds are treated in the class.